File: client/src/app/_models/storyParams.ts
import { User } from "./user";

export class StoryParams{
    genre :string;
    author : string;
    storyType:string;
    language:string;
    search:string;
    pageNumber = 1;
    pageSize = 10;
    // topViews = 'views'
    // topRate = 'rating'
    orderBy = 'rating';
    constructor(user?:User){
        this.genre ='All';
        this.author='';
        this.storyType='novel';
        this.language='All';
        this.search='';
    }
}

File: API/Helpers/StoryParams.cs
namespace API.Helpers
{
    public class StoryParams :PaginationParams
    {
        // public string CurrentUsername { get; set; }
        public string Genre { get; set; }
        public string Author { get; set; }
        public string StoryType { get; set; }
        public string OrderBy { get; set; } = "created";
        public string OrderByViews { get; set; } ="weekly";
        public string Language { get; set; }
        public string Search { get; set;}
        public int UserId { get; set; }
    }
}